Добавляем в tpl кастомные параметры из вызова

Alexij2 opened the pull request:
### Что оно делает?
Добавляем в tpl кастомные параметры из вызова сниппета, чтобы ими пользоваться в самой tpl-ке вывода.
Уже есть все свойства к wrapp обертке вывода, но это далеко не удобно.

в msProduct, Добавление свойств, как это сделано в pdoResources Если считаете что нужно по-другому - поправьте, но думаю было бы хорошо, чтобы логика поведения была одинакова как в pdoToolse сниппетах так и msProduct.

### Связанные проблема(ы)/PR(ы)
Дополнение к
#711

Alexij2 committed:
Добавляем в tpl кастомные параметры из вызова

Дополнение к 
#711
в msProduct, Добавление свойств, как это сделано в pdoResources
Если считаете что нужно по-другому - поправьте, но думаю было бы хорошо, чтобы логика поведения была одинакова как в pdoToolse сниппетах так и msProduct.

biz87 commented on core/components/minishop2/elements/snippets/snippet.ms_products.php:
```diff
@@ -204,6 +204,24 @@
+    $addplace = $tmprops = [];
+    if (isset($this) && $this instanceof modSnippet && $this->get('properties')) {
+        $tmprops = $this->get('properties');
+    }
```
Перенос строки в этом месте не рекомендован.  Должен быть один пробел вместо переноса

biz87 commented on core/components/minishop2/elements/snippets/snippet.ms_products.php:
```diff
@@ -204,6 +204,24 @@
+        }
+    }
+    
+    
```
Лишний перенос строки

biz87 commented on core/components/minishop2/elements/snippets/snippet.ms_products.php:
```diff
@@ -224,7 +242,7 @@
         $opt_time_start = microtime(true);
         $options = $modx->call('msProductData', 'loadOptions', array($modx, $row['id']));
-        $row = array_merge($row, $options);
+        $row = array_merge($row, $options,$addplace);
```
Между параметрами рекомендован пробел

Alexij2 committed:
Исправил ошибки.

Исправил, подправил.
Предлагаю еще для $outer параметры всовывать сразу на уровне output. как в pdoToolse.
Думаю, проблем не должно возникнуть.
спасибо!

Alexij2 committed:
мелкая правка.

Забыл указать msProduct.
